packages/images-plugin/lib/compare.js

```diff
@@ -90,4 +107,33 @@ export {
 
 /**
  * @typedef {import('@appium/opencv').OccurrenceResult} OccurrenceResult
+ * @typedef {import('@appium/opencv').SimilarityOptions} SimilarityOptions
+ * @typedef {import('@appium/opencv').MatchingOptions} MatchingOptions
+ * @typedef {import('@appium/opencv').SimilarityResult} SimilarityResult
+ * @typedef {import('@appium/opencv').MatchingResult} MatchingResult
+ */
+
+/**
+ * @template {boolean} Multiple
+ * @typedef {import('@appium/opencv').OccurrenceOptions<Multiple>} OccurrenceOptions
+
+ */
+
+/**
+ * @typedef {typeof GET_SIMILARITY_MODE} GetSimilarityMode
+ * @typedef {typeof MATCH_FEATURES_MODE} MatchFeaturesMode
+ * @typedef {typeof MATCH_TEMPLATE_MODE} MatchTemplateMode
+ * @typedef {GetSimilarityMode|MatchFeaturesMode|MatchTemplateMode} CompareMode
+ */
+
+/**
+ * @template {CompareMode} Mode
+ * @template {boolean} Multiple
+ * @typedef {Mode extends GetSimilarityMode ? SimilarityOptions : Mode extends MatchFeaturesMode ? MatchingOptions : Mode extends MatchTemplateMode ? OccurrenceOptions<Multiple> : never} CompareImagesOptions
```

Comment:
this line is long and hard to read

Reply:
short of making new typedefs, there's nothing to be done, because linebreaks break the type definition.